### PR TITLE
Removing Apex Connectivity Client

### PR DESCRIPTION
**Pull request by anathoodell**
# Description
A few sentences describing the overall goals of the pull request's commits.

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
|   https://github.com/dell/csm/issues/1484|

# Checklist:

- [ ] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [ ] I have verified that new and existing unit tests pass locally with my changes
- [ ] I have not allowed coverage numbers to degenerate
- [ ] I have maintained at least 90% code coverage
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have maintained backward compatibility

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -225,17 +225,19 @@ func main() {
...
+			K8sClient:     k8sClient,
+			Log:           log,
+			Scheme:        mgr.GetScheme(),
+			EventRecorder: recorder,
+			Config:        operatorConfig,
+		}).SetupWithManager(mgr, expRateLimiter, 1); err != nil {
+			setupLog.Error(err, "unable to create controller", "controller", "ApexConnectivityClient")
+			os.Exit(1)
+		}
+	*/
```
Comment (francis-nijay) [inline main.go:240]: Can we have this commented code removed?

**Review by francis-nijay — COMMENTED** [review]
There are some yaml lint errors

File: main.go (hunk 1 above)
Comment (anathoodell) [reply main.go]: removed.

File: pkg/utils/utils.go [hunk 2]
```diff
@@ -364,29 +364,6 @@ func ModifyCommonCR(YamlString string, cr csmv1.ContainerStorageModule) string {
...
-	for _, env := range cr.Spec.Client.Common.Envs {
-		if env.Name == "KUBELET_CONFIG_DIR" {
-			path = env.Value
-			break
-		}
-	}
-	YamlString = strings.ReplaceAll(YamlString, KubeletConfigDir, path)
-
-	return YamlString
-}
```
Comment (ChristianAtDell) [inline pkg/utils/utils.go:388]: Is this related to ACC only? 'CommonCRs' makes me worry that this functionality could be in use elsewhere.

File: pkg/utils/utils.go [hunk 3]
```diff
@@ -1300,29 +1263,9 @@ func GetNamespaces(ctx context.Context, ctrlClient crclient.Client) ([]string, e
...
-			csmv1.DreadnoughtClient, accConfigVersion, BrownfieldManifest)
-		if err = BrownfieldOnboard(ctx, brownfieldManifestFilePath, clientNamespace, ctrlClient, isDeleting); err != nil {
-			logInstance.Error(err, "error creating role/rolebindings")
-			return err
-		}
-	}
-	return nil
-}
-
+// 2nd parameter is an ApexCC cr
```
Comment (ChristianAtDell) [inline pkg/utils/utils.go:1266]: Should this entire comment block be removed? It is related only to ACC, which we are trying to remove.

File: pkg/utils/utils.go [hunk 4]
```diff
@@ -1300,29 +1263,9 @@ func GetNamespaces(ctx context.Context, ctrlClient crclient.Client) ([]string, e
...
 // CreateBrownfieldRbac creates the role and rolebindings
+/*
```
Comment (ChristianAtDell) [inline pkg/utils/utils.go:1268]: Should we remove this code instead of commenting it out? It seems like all of our RBAC work in this repo was tied to ACC which is being removed.

File: tests/e2e/steps/steps_def.go [hunk 5]
```diff
@@ -91,6 +91,7 @@ var correctlyAuthInjected = func(cr csmv1.ContainerStorageModule, annotations ma
 }
 
 // GetTestResources -- parse values file
+// TBD: remove apex param from here and caller
```
Comment (ChristianAtDell) [inline tests/e2e/steps/steps_def.go:94]: Is there a blocker that is preventing us from removing this param now?

File: pkg/utils/utils.go (hunk 2 above)
Comment (anathoodell) [reply pkg/utils/utils.go:388]: ModifyCommonCRs was only called via acc_controller.go and commonconfig.go GetAccController, both of which are gone, so it's fine to remove this as it is unused dead code.

File: pkg/utils/utils.go (hunk 3 above)
Comment (anathoodell) [reply pkg/utils/utils.go:1266]: yes, removed.

File: pkg/utils/utils.go (hunk 4 above)
Comment (anathoodell) [reply pkg/utils/utils.go]: yes, removed.

**Comment by anathoodell** [comment]
The commits in this PR are merged into pub/krv-28556 branch, and a PR for this is available https://github.com/dell/csm-operator/pull/721. As a result, this PR is being closed.
